tools/eliminate-duplicate-functions.js

```diff
@@ -0,0 +1,489 @@
+///////////////////////////////////////////////////////////////////////////////////////////////
+// Eliminate-Duplicate-Functions.js
+//
+// This is a Javascript file that is used to post-process an Emscripten transpiled JS file.
+// It will remove all the duplicate functions from the generated ASM. In its current form,
+// the input JS file is expected to be an Emscripten generated file of the form:
+//
+// <PRE-ASM-JS-SHELL>
```

Comment:
Yes, you're right - this comment is a bit out of date. I'll update the comment to reflect this. Although we also do pass in the ASM 'shell' which is the part contained within the asm variable declaration.